src/main/java/com/crowsofwar/avatar/common/bending/earth/AbilityWall.java

```diff
@@ -89,91 +90,217 @@ public void execute(AbilityContext ctx) {
 
 			abilityData.addXp(SKILLS_CONFIG.wallRaised);
 
-			// Down 1 block so that we actually get a block...
-			BlockPos lookPos = entity.getPosition().down().offset(cardinal, reach);
-			EntityWall wall = new EntityWall(world);
-
-			Block lookBlock = world.getBlockState(lookPos).getBlock();
-			if (lookBlock == Blocks.TALLGRASS) {
-				lookPos = lookPos.down();
-			} else if (lookBlock == Blocks.DOUBLE_PLANT) {
-				lookPos = lookPos.down(2);
+			// The range should increase the higher the level is.
```

Comment:
Culd all be replaced with
```suggestion
			reach += abilityData.getLevel();
```